OpenRA.Mods.Common/UpdateRules/UpdateUtils.cs

```diff
@@ -237,6 +237,12 @@ public static void Save(this YamlFileSet files)
 					file.Item1.Update(file.Item2, Encoding.UTF8.GetBytes(file.Item3.WriteToString()));
 		}
 
+		/// <summary>Only sets the yaml key of the node. Does not preserve the @suffix!</summary>
+		public static void SetKey(this MiniYamlNode node, string newKey)
```

Comment:
`RenameKeyPreservingSuffix` should be fine for most of the things you are using this for.  Do we really need another version?  If we really do, then this should be named `RenameKey` rather than `SetKey` for consistency.

Reply:
I used this for https://github.com/OpenRA/OpenRA/pull/15010/commits/b61bf340f1b6aeea21c31cfb5e68d6cd3984d2ab#diff-0bebab4e0f73ce1d3baeb516eb74e974R35 (`requiresPower.SetKey("GrantConditionOnPowerState@LOWPOWER");`). I'm not sure if using `RenameKeyPreservingSuffix` works in that case, as one could end up with a second suffix. I can think of another way if we don't want a second rename method. (Btw, I named it `SetKey` with the intention of making clear that the method doesn't preserve the suffix but overwrites it entirely.)

Reply:
I'm not sure what the advantage is over setting the desired key directly (`requiresPower.Key("GrantConditionOnPowerState@LOWPOWER";`)?
It doesn't even save anything in terms of line length (`Set` + `()` brackets = 5 bytes, ` = ` = 3 bytes).